### ISSUE TITLE
Release a quick ggplot2 3.0.1?

### ISSUE THREAD

**Issue by clauswilke** [tidyverse/ggplot2#2831]
I would like to open a conversation about the feasibility of releasing a bugfix release 3.0.1 relatively soon. If I look at the merged changes so far (https://github.com/tidyverse/ggplot2/blob/master/NEWS.md), they're mostly things that should have been caught for 3.0.0 but weren't. And in particular, issues #2363 (new graphics device with `ggsave()`) and #2788 (tidy eval for secondary axis) will be frustrating for users of 3.0.0.

I realize how much work it is to prep a release. At the same time, once we start merging code that makes more fundamental changes to the code base, the window of opportunity for a pure bug-fix release is essentially closed. So, a plan to prioritize work might be good. E.g., only merge bug fixes until the end of September, prep a 3.0.1 release then, and start merging more ambitious work after 3.0.1.

I assume we don't want to go to a development model where we're working on a minor and a major release at the same time, in different branches, but that would be an option as well of course.

**Comment by hadley** [comment]
I like the idea of planning for a minor release mid-October. I think as long as we focus on bug fixes and backward compatible changes, we only need two weeks for the release window.

(So far the only big change is "Data is no longer internally reordered when faceting" (which is missing an issue/PR number) — it seems likely that will change existing plots, but only in a way that makes them more correct)

**Comment by clauswilke** [comment]
The "no data reordering" commit belongs to PR #2694. There was no associated issue that was closed. If the rule is to list PR numbers when there is no issue then I can add that number to the NEWS file.

I also think that I can separate the pending PR #2821 into a pure bug-fix component and a component that cleans up the API. Then we could merge the bug-fix component now and leave the API cleanup for later.

**Comment by hadley** [comment]
Yeah, the policy is to include PR numbers if no issue (sorry for not catching that at the time). The policy is described on http://style.tidyverse.org/news.html but it was a bit buried - I just reorganised to hopefully make it more clear. My goal is to make sure that we can easily link to a section of the style guide for any common violations.

Proposed plan for #2821 sounds good.

**Comment by clauswilke** [comment]
I've gone over open PRs and issues to see which could/should be completed before a 3.0.1 release. 

Among my own PRs, it would be nice if we could merge these:

- #2881 Improved parsing for labels in `coord_sf()`. Ready to be merged, I think.
- #2849 Make graticule labeling configurable. It may need one more round of discussion about the ideal interface; also, needs to be rebased after #2881 is merged.
- #2813 Make `preserve = "single"` more robust in `position_dodge()`. Needs a news item; otherwise is fine I think.

@dpseidel has two that may make sense for 3.0.1:

- #2806 Implement `sec.axis()` for date, time, and datetime scales.
- #2792 Documentation updates for CII badging requirements. The PR doesn't address all badging requirements yet but the documentation changes in the PR seem appropriate and could be merged.

And there are two more that may be worthwhile, but I don't feel qualified to evaluate them:

- #2808 Document now unused argument.
- #2807 Force arguments that are used in closures.

I also looked over the open issues but didn't see any that I'd consider must-fix for 3.0.1.

**Comment by hadley** [comment]
Ok, I've kick all of those issues along.

@clauswilke are you willing to official manage this release? (You're already doing most of the work). If so, can you please create a new issue from the template at https://github.com/r-lib/usethis/issues/338, and then start checking bits off? I'll be travelling next week, but I'll do my best to respond to any questions/queries as quickly as possible.

**Comment by clauswilke** [comment]
I opened a new issue (#2890) and will close this one. Somebody else will have to run the revdepcheck, I think. I don't have a desktop set up where I could let this chug along for however many hours it takes.

**Comment by hadley** [comment]
Just ping @topepo — he has the dubious honour of being the official ggplot2 revdep runner

**Comment by clauswilke** [comment]
@topepo We're not there yet. I'll let you know when we're ready. Thanks!

**Comment by josesho** [comment]
Is #2788 going to be part of ggplot 3.0.1? If not, can I request that it be included? (Working on fixing mission-critical function, and this breaks it in a massive way.)
